wp-admin/js/widgets/text-widgets.js

```diff
@@ -32,25 +32,87 @@ wp.textWidgets = ( function( $ ) {
+				throw new Error( 'Missing options.el' );
+			}
+
 			Backbone.View.prototype.initialize.call( control, options );
 
-			if ( ! control.el ) {
-				throw new Error( 'Missing options.el' );
+			/*
+			 * Create a container element for the widget control fields.
+			 * This is inserted into the DOM immediately before the the .widget-content
+			 * element because the contents of this element are essentially "managed"
+			 * by PHP, where each widget update cause the entire element to be emptied
+			 * and replaced with the rendered output of WP_Widget::form() which is
+			 * sent back in Ajax request made to save/update the widget instance.
+			 * To prevent a "flash of replaced DOM elements and re-initialized JS
+			 * components", the JS template is rendered outside of the normal form
+			 * container.
+			 */
+			control.fieldContainer = $( '<div class="text-widget-fields"></div>' );
+			control.fieldContainer.html( wp.template( 'widget-text-control-fields' ) );
+			control.widgetContentContainer = control.$el.find( '.widget-content:first' );
+			control.widgetContentContainer.before( control.fieldContainer );
+
+			control.fields = {
+				title: control.fieldContainer.find( '.title' ),
+				text: control.fieldContainer.find( '.text' )
+			};
+
+			// Sync input fields to hidden sync fields which actually get sent to the server.
+			_.each( control.fields, function( fieldInput, fieldName ) {
+				fieldInput.on( 'input change', function updateSyncField() {
+					var syncInput = control.widgetContentContainer.find( 'input[type=hidden].' + fieldName );
+					if ( syncInput.val() !== $( this ).val() ) {
+						syncInput.val( $( this ).val() );
+						syncInput.trigger( 'change' );
+					}
+				});
+
+				// Note that syncInput cannot be re-used because it will be destroyed with each widget-updated event.
+				fieldInput.val( control.widgetContentContainer.find( 'input[type=hidden].' + fieldName ).val() );
+			});
+		},
+
+		/**
+		 * Update input fields from the sync fields.
+		 *
+		 * This function is called at the widget-updated and widget-synced events.
+		 * A field will only be updated if it is not currently focused, to avoid
+		 * overwriting content that the user is entering.
+		 *
+		 * @returns {void}
+		 */
+		updateFields: function updateFields() {
+			var control = this, syncInput;
+
+			if ( ! control.fields.title.is( document.activeElement ) ) {
+				syncInput = control.widgetContentContainer.find( 'input[type=hidden].title' );
+				control.fields.title.val( syncInput.val() );
+			}
+
+			syncInput = control.widgetContentContainer.find( 'input[type=hidden].text' );
+			if ( control.fields.text.is( ':visible' ) ) {
+				if ( ! control.fields.text.is( document.activeElement ) ) {
+					control.fields.text.val( syncInput.val() );
+				}
+			} else if ( control.editor && ! control.editorFocused && syncInput.val() !== control.fields.text.val() ) {
+				control.editor.setContent( wp.editor.autop( syncInput.val() ) );
```

Comment:
@timmyc In the customizer, there is already logic in place in core to prevent updates from overriding content that the user is entering: https://github.com/WordPress/wordpress-develop/blob/656718a645632771207b1ecb54a0c49e4b06033a/src/wp-admin/js/customize-widgets.js#L1221-L1227

On the widgets admin screen, on the other hand, it is already “the way it works” for fields to be blown away after hitting save. So I don't think we're introducing anything new.